### PR TITLE
React: Fix correct directory for _redirects (Netlify only)

### PR DESCRIPTION
**Pull request by MaoShizhong**
## Because
Netlify and Vite copies public/ into dist/ when building, so _redirects needs to go there.
If _redirects is in the root of the project, it is not picked up in build and so will not redirect.
Cloudflare does not require a _redirects at all (tested) and the docs state that if a top-level
404.html is not included in the build, it will assume SPA and route accordingly.

## This PR
- Fix with correct directory for Netlify `_redirects` (`public/` as opposed to project root)
- Remove Cloudflare pages mention as no extra step is required unlike Netlify/Vercel
- Remove Cloudflare docs mention at the bottom for the same reason

## Issue
No open issue

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by MaoShizhong** [comment]
Just for future reference, since this is something that is in regards to behaviour that should probably be checked and verified before approving such changes, would something like this be better off as an issue first, then simply saying I wish to be assigned to the issue if verified and agreed?

**Comment by Lofty-Brambles** [comment]
It was meant to be "at the root of the `/public` directory 😅
With cloudflare, the decision was intentional- It's better to have explicit instructions, when all it'd take is one extra line. Otherwise, it'd also be great to mention the fact that you do not need to do anything else, with it. On the side of caution, better to mention it in case the default behaviour changes.

**Comment by MaoShizhong** [comment]
I can push another commit that brings back the Cloudflare docs, but puts a separate Cloudflare pages mention saying that you currently do not need to do anything extra like add a _redirects unlike Netlify or Vercel's instructions?
Since if Cloudflare pages behaviour does change, there's no guarantee it will require the same method as Netlify (just like Vercel is different)

File: react/the_react_ecosystem/project_shopping_cart.md [hunk 1]
```diff
@@ -37,6 +37,8 @@ By now you've come far from your React-baby days, with tools like routers and te
         }
         ~~~
 
-    You can learn more about how these work at the [Netlify](https://docs.netlify.com/routing/redirects/) and [Vercel](https://vercel.com/docs/frameworks/vite#using-vite-to-make-spas) docs respectively.
+      3. **Cloudflare Pages**: As of the time of writing, unlike Netlify and Vercel, no additional steps are required as the default behaviour will correctly handle single page applications.
+
+    You can learn more about how these work at the  [Netlify](https://docs.netlify.com/routing/redirects/), [Vercel](https://vercel.com/docs/frameworks/vite#using-vite-to-make-spas) and [Cloudflare](https://developers.cloudflare.com/pages/platform/serving-pages/), docs respectively.
```
Comment (ManonLef) [inline react/the_react_ecosystem/project_shopping_cart.md]:
```suggestion
    You can learn more about how these work at the [Netlify](https://docs.netlify.com/routing/redirects/), [Vercel](https://vercel.com/docs/frameworks/vite#using-vite-to-make-spas) and [Cloudflare](https://developers.cloudflare.com/pages/platform/serving-pages/) documentation respectively.
```

Comment (ManonLef) [reply react/the_react_ecosystem/project_shopping_cart.md]: removed an extra space and I think the last comma isn't needed. More descriptive links would be even better if you can come up with something :)

Thanks for adding these documentation links!

**Review by ManonLef — CHANGES_REQUESTED** [review]
see the comments regarding the edits I proposed. Perhaps you can come up with a better link description?

**Comment by ManonLef** [comment]
I also believe the nesting inside the numbered lists has a space too many according to the [Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#lists). If you want to give that a look as well, that would be great.

**Comment by MaoShizhong** [comment]
I caught the extra , at the end but missed the issues with spacing/indentation! These have been fixed and I can't seem to find any other issues that don't follow the style guide/what was originally there.
I reworded the cloudflares description to be more in line with the wording for Netlify and Vercel, hopefully it's a bit more unified now.

File: react/the_react_ecosystem/project_shopping_cart.md [hunk 2]
```diff
@@ -37,6 +37,8 @@ By now you've come far from your React-baby days, with tools like routers and te
         }
         ~~~
 
-    You can learn more about how these work at the [Netlify](https://docs.netlify.com/routing/redirects/), [Vercel](https://vercel.com/docs/frameworks/vite#using-vite-to-make-spas) and [Cloudflare Pages](https://developers.cloudflare.com/pages/platform/redirects/) docs respectively.
+    3.  **Cloudflare Pages**: As of the time of writing, unlike Netlify and Vercel, no additional steps are required as the default behaviour will allow `react-router-dom` to correctly handle redirects.
+
+    You can learn more about how these work at the [Netlify](https://docs.netlify.com/routing/redirects/), [Vercel](https://vercel.com/docs/frameworks/vite#using-vite-to-make-spas) and [Cloudflare](https://developers.cloudflare.com/pages/platform/serving-pages/) documentation respectively.
```
[the remark this answers is not in the document]
Comment (MaoShizhong) [reply react/the_react_ecosystem/project_shopping_cart.md:42]: Sure thing! I kept what was originally there but I'll have a look at making it more descriptive for sure!

**Review by ManonLef — APPROVED** [review]
Thanks for the update on the links! Much better and great way of rewriting them.

**Comment by ManonLef** [comment]
I will leave the final review to someone that knows more about the contents of the lesson.

**Comment by yuliana-r** [comment]
Had the same issue today and took me a while to figure it out! Could this please be looked into so that this can be avoided by anyone working on their shopping cart project?

Thanks!

**Comment by ManonLef** [comment]
@TheOdinProject/react could someone have a look at this please :)

**Review by 01zulfi — APPROVED** [review]
lgtm :rocket:

**Pull request merged**